.github/workflows/build-and-push-components.yml

```diff
@@ -22,6 +22,35 @@ on:
         type: string
         description: 'Name of the Docker Bake group of targets'
         default: default
+  workflow_dispatch:
+    inputs:
+      git_ref:
+        required: false
+        type: string
+        description: 'The branch, tag or SHA to build.'
+        default: ''
+      image_tag:
+        required: true
+        type: string
+        description: 'Image tag to use.'
+      registry_name:
+        required: false
+        type: string
+        description: 'Name of the registry.'
+        default: ghcr.io/openclarity
+      suffix:
+        required: false
+        type: string
+        description: 'Suffix to append to the image name'
+        default: ""
+      bake-group:
+        required: false
+        type: string
+        description: 'Name of the Docker Bake group of targets'
+        default: default
+
+permissions:
+  packages: write
```

Comment:
This workflow should have one responsibility, now it has two, I believe we should keep the manual workflow, and have these specified there.

Reply:
It is a perfectly valid use case to trigger a workflow on multiple events: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-activity-types-and-filters-with-multiple-events, that does not mean it would have "multiple responsibilities".

If you are referring to the SRP, what exactly could be the reasons for the changes that are different in the two cases? Clarifying this would help to decide whether if it is worth extracting the manual job into a separate file, the "it should have one responsibility" is quite general and not so actionable.

Reply:
> BTW, Your previous implementation of the manual build and push was something very similar to what I implemented.

My previous implementation was a quick fix for a problem that an external team had, not a well thought-out solution.

> If you look at any of the workflow files right now, it's immediately clear what it does, and how that thing is done.

If you look at this at this file, it still does the same, based on the inputs it will build and push the vmclarity components. The only change that now it can be triggered on multiple events. The inputs are the same, everything is the same.

Either way, I will put the workflow_dispatch into a separate file.